src/main/java/org/kitteh/irc/client/library/util/StringUtil.java

```diff
@@ -157,4 +158,51 @@ public static String toLowerCase(@Nonnull Client client, @Nonnull String input)
         Sanity.nullCheck(input, "Input cannot be null");
         return client.getServerInfo().getCaseMapping().toLowerCase(input);
     }
+
+    /**
+     * Converts an IRC wildcard (*, ?) to a regular expression pattern.
+     *
+     * @param wildcardExpression expression
+     * @return a pattern
+     * @throws IllegalArgumentException if wildcardExpression is null
+     */
+    @Nonnull
+    public static Pattern wildcardToPattern(@Nonnull String wildcardExpression) {
+        Sanity.nullCheck(wildcardExpression, "Wildcard expression cannot be null");
+
+        StringBuilder builder = new StringBuilder(wildcardExpression.length());
+        for (char character : wildcardExpression.toCharArray()) {
+            switch (character) {
+                case '*':
+                case '?':
+                    builder.append('.').append(character); // * to .* and ? to .?
```

Comment:
That's not how a `?` works, a ? translates to regex `.`,  a `*` translates to regex `.*`

Regex `.?` would be "zero or one of any", When in reality `?` matches exactly one of any.

Case in point:

```
+b ????*!*@*
````

Bans all four-or-more letter nicks.

With your regex, this would ban all 0 or more letter nicks.